Laba3

PolinaKapustina opened the pull request.

PolinaKapustina committed:
Add files via upload

PolinaKapustina committed:
New

PolinaKapustina committed:
Надеюсь всё обновится

PolinaKapustina committed:
Delete utmatrix.h

PolinaKapustina committed:
Delete test_tmatrix.cpp

PolinaKapustina committed:
Delete sample_matrix.cpp

PolinaKapustina committed:
Delete test_tvector.cpp

PolinaKapustina committed:
Add files via upload

PolinaKapustina committed:
Add files via upload

PolinaKapustina committed:
Add files via upload

PolinaKapustina committed:
Add files via upload

ArisovaAnastasia commented:
Есть замечания:
Программа падает на примере "5+5" в строке 313 файла "arithmetic.cpp", другие примеры пока не проверяла. 
При вводе выражения с переменными, например "4*х" программа также падает. 
Также сейчас не проходят некоторые тесты, необходимо это исправить. 
Замечания по отчету: в раздел Приложение необходимо добавить код нескольких функций, реализующих основные алгоритмы.
Примечание: у меня не собралась программа. После замены строки 14 в файле "stack.h" ("int MaxStackSize = 100;") на строку "#define MaxStackSize 100" все собралось

PolinaKapustina committed:
Временное обновление

Исправила часть ошибок, проверить пока не могу. так как не нахожу как поменять запускаемый файл для отладки (запускает gtest, вместо нужного). Надеюсь, в ближайшее время найду как менять и отлажу всё до конца.

ArisovaAnastasia commented:
Чтобы назначить проект автозапускаемым, нужно нажать на него правой кнопкой мыши и выбрать "Set a StartUp project"

PolinaKapustina committed:
Лабораторная, где, наверное, не должно быть ошибок

ArisovaAnastasia commented:
Сейчас некоторые тесты падают, необходимо это исправить. Также есть примеры, которые работают неверно:
"х-6" программа падает
"(12-12" программа падает
"1.2-.13" программа падает
Все выбрасываемые исключения должны обрабатываться программой, и выводиться сообщение об ошибке пользователю.

PolinaKapustina commented:
Здравствуйте, Анастасия Николаевна, пытаюсь исправить найденные ошибки, но не могу понять как это сделать. В CorrectionChecker (15-56 строка) исключения throw не выдаются и возникает не обработанное исключение. Почему так происходит не могу понять. Обнаружила, что программа падает даже при вводе символа, не относящегося к допустимым, хотя прописала, чтобы в этом случае выдавало ошибку. Не понимаю что с этим можно сделать. Пыталась с нуля переписывать отрывок, но ничего не поменялось.

ArisovaAnastasia commented:
В функции CorrectionChecker (строка 25) почему-то не указан символ "." как разрешенный, из-за этого при вводе выражения "1.3" программа выбрасывает исключение.
Также все исключения необходимо обрабатывать, чтобы программа не падала, а выводила сообщение об ошибке и продолжала работать. Для этого нужно использовать конструкцию try{...} catch {...}.
Попробуйте еще раз проверить логику работы функции CorrectionChecker() и протестировать ее. Если возникнут примеры, где Вы не понимаете почему она не работает - пишите. Желательно указывать конкретный пример, на котором не работает.

PolinaKapustina commented:
Хорошо, благодарю за помощь, постараюсь исправить указанные проблемы.

PolinaKapustina committed:
Обновление

Теперь ошибки выводятся корректно, из проблем даже после вывода ошибки программа выводит результат операций (которые не должна считать). Пример: 1..3 -- есть сообщение об ошибке, но выводит 1. Тесты пока что в процессе окончательной доработки (так же как и программа в принципе)

PolinaKapustina committed:
Всё работает

Вроде бы всё начало работать, тесты проходятся, в  cath используется exit  для безопасного выхода.

PolinaKapustina commented:
Постаралась исправить проблемы. Вроде бы всё работает хорошо.

ArisovaAnastasia commented:
Проверьте, пожалуйста, вы точно обновили все файлы до новой версии? Сейчас я не вижу существенных изменений, например, добавлении конструкции try-catch

PolinaKapustina committed:
Обновись пожалуйста

PolinaKapustina committed:
Вроде теперь точно всё

PolinaKapustina commented:
Обновила снова, надеюсь в этот раз всё что нужно

PolinaKapustina committed:
Add files via upload

PolinaKapustina commented:
Перенесла где могла всё в try/cath, исключением стала программа PolishNotationCalculate(). Пыталась реализовать непрерывное выполнение программы, с новым вводом в случае ошибки, но каждый раз сталкиваюсь с различными ошибками, идеи исправления которых на данный момент кончились. Сейчас при запуске выдаёт "ucrtbased.pdb содержит отладочную информацию, необходимую для поиска исходного кода для модуля ucrtbased.dll", что с этим делать так и не поняла. Все тесты проходятся корректно, переписывать их не пришлось. Осталось разобраться как сделать корректной и соответствующей требованиям работу main_arithmetic.cpp

PolinaKapustina committed:
чтобы наверняка

ArisovaAnastasia commented:
1. Указанная Вами ошибка возникает в строчке 20 функции main. Конструктор класса принимает переменную типа string, а вы передаете 0. Нужно либо передавать созданную выше строку s, либо указывать "0". 
2. Сейчас при запуске программы даже на примере "5+5" она падает. Вы создаете в строке 25 объект v, он является локальным, так как создан в теле цикла while(). Вы работаете с ним, обрабатываете выражение и сохраняете его вид в ОПЗ. Затем блок while заканчивается, и данный объект удаляется. Затем в 47 строке вы пытаетесь вычислить выражение сохраненное в объекте с таким же названием v, но созданным в строке 20. В этом объекте нет никакой информации, вы его не инициализировали и соответственно при попытке его вычислить программа падает.
3. Нужно еще раз продумать логику работы приложения. Также сейчас есть дублирование кода. В блок try можно помещать вызов несколько функций.

PolinaKapustina committed:
Обновление

PolinaKapustina commented:
Конструктор класса с объектов v  были отправлены в цикл, Try теперь обрабатывает 2 функции, вторая переписана под ошибку типа int. Дублирование, кажется, убрано, если правильно поняла в чём беда

ArisovaAnastasia commented:
Теперь некорректные выражения обрабатываются верно. Но программа по прежнему обрабатывает только одно выражение и завершается. Необходимо запрашивать пользователя, хочет ли он ввести еще одно выражение и считывать его ответ в какую-нибудь переменную. И добавить еще один цикл while, чтобы в зависимости от значения этой переменной либо вся работа приложения повторялась, либо программа завершалась.

PolinaKapustina committed:
Обновление 1

PolinaKapustina committed:
Обновление1

PolinaKapustina committed:
Add files via upload

PolinaKapustina committed:
End

PolinaKapustina commented:
Добавила повторное использование программы на выбор пользователя.

ArisovaAnastasia commented:
Лабораторная работа принята

PolinaKapustina commented:
Благодарю!
